src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs

```diff
@@ -50,19 +52,19 @@ private static Span<uint> PowCore(Span<uint> value, int valueLength, Span<uint>
             Debug.Assert(value.Length == temp.Length);
 
             result[0] = 1;
-            int bitsLength = 1;
+            int resultLength = 1;
 
             // The basic pow algorithm using square-and-multiply.
             while (power != 0)
             {
                 if ((power & 1) == 1)
-                    bitsLength = MultiplySelf(ref result, bitsLength, value.Slice(0, valueLength), ref temp);
+                    resultLength = MultiplySelf(ref result, resultLength, value.Slice(0, valueLength), ref temp);
                 if (power != 1)
                     valueLength = SquareSelf(ref value, valueLength, ref temp);
                 power >>= 1;
             }
 
-            return result;
+            return result.Slice(0, resultLength);
```

Comment:
Notably, this function and functions that use it were not touched by https://github.com/dotnet/runtime/pull/74112. However, if you think the same bug was already present here before that PR was merged, then this is fine to include.

Reply:
The concern is that a similar bug could exist. The computed `resultLength` should be correct and so slicing should not be harmful and will help avoid similar issues as to what we already saw.